File: ios/equipapp/equipapp/Table/Recording.swift
//
//  Recording.swift
//  equipapp
//
//  Created by Nathan Sinclair on 5/12/18.
//  Copyright © 2018 Nathan Sinclair. All rights reserved.
//

import UIKit

class Recording: NSObject {
    
    var recordingName :String
    var date: Date
    var dateStringForTable: String = ""
    var downloadURL: URL
    var speaker: String
    init(newRecordingName: String, newDate: Date, url: String, newSpeaker: String) {
      
        self.recordingName = newRecordingName
        self.date = newDate
        let urlObject = URL(string: url)
        self.downloadURL = urlObject!
        self.speaker = newSpeaker
        
    }
    
    func createDate() {
        let formatter = DateFormatter()
        formatter.dateFormat = "dd/mm/YYYY"
        self.dateStringForTable = formatter.string(from: date)
    }

}

File: android/app/src/main/java/com/nathan/equipapp/assets/Recording.kt
package com.nathan.equipapp.assets

import android.util.Log
import org.threeten.bp.DateTimeUtils
import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneId
import org.threeten.bp.ZoneOffset
import java.util.*

class Recording {
    var speaker: String
    var name: String
    var date: Date
    var url: String
    var formattedDate: String? = null
    val TAG = this.javaClass.simpleName

    constructor(newSpeaker: String, newName: String, newDate: Date, newUrl: String) {
        this.speaker = newSpeaker
        this.name = newName
        this.date = newDate
        this.url = newUrl
        createDate()

    }

    fun createDate() {

        var formatter = org.threeten.bp.format.DateTimeFormatter.ofPattern("eee, d MMMM YYYY")
        val aucklandTime = LocalDateTime.ofInstant(DateTimeUtils.toInstant(date), ZoneId.of("Pacific/Auckland"))
        formattedDate = aucklandTime.format(formatter)

    }
}